llvm/lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -1082,8 +1088,44 @@ bool X86RegisterInfo::getRegAllocationHints(Register VirtReg,
       VirtReg, Order, Hints, MF, VRM, Matrix);
 
   unsigned ID = RC.getID();
-  if (ID != X86::TILERegClassID)
+
+  if (ID != X86::TILERegClassID) {
+    if (!VRM || DisableRegAllocNDDHints)
+      return BaseImplRetVal;
+
+    // Add any two address hints after any copy hints.
+    SmallSet<unsigned, 4> TwoAddrHints;
```

Comment:
Wouldn't it introduce indeterministic behavior since you push its elements into a vector? (AMX code seems a bug too.)

Maybe we should use `SmallSetVector` here.

Reply:
Never mind. I see you iterate on `OrderReg`.